diagnostics-app/src/app/diagnostics/diagnostics-dashboard/diagnostics-dashboard.component.ts

```diff
@@ -1,10 +1,65 @@
-import { Component } from '@angular/core';
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview Defines the Dashboard Component in Diagnostics.
+ * This is the root layout component which holds the diagnostics dashboard.
+ * Imported by diagnostics.module.ts
+ */
+
+import { Component, OnInit } from '@angular/core';
+
+import { GetAvailableRoutinesResponse, RoutineType } from '@common/dpsl';
+import { DiagnosticsResponse, ResponseErrorInfoMessage } from '@common/message';
+import { DiagnosticsService } from 'src/app/core/services/diagnostics.service';
 
 @Component({
   selector: 'app-diagnostics-dashboard',
   templateUrl: './diagnostics-dashboard.component.html',
   styleUrls: ['./diagnostics-dashboard.component.css']
 })
-export class DiagnosticsDashboardComponent {
+export class DiagnosticsDashboardComponent implements OnInit {
+
+  private _availableRoutines?: RoutineType[]; // the available routines response
+  private _error?: ResponseErrorInfoMessage; // the error message received
+
+  // Array of diagnostics routine names that will be displayed
+  private _visibleRoutines: RoutineType[] = [
+    RoutineType.ac_power,
+    RoutineType.audio_driver,
+    RoutineType.battery_charge,
+    RoutineType.battery_discharge,
+    RoutineType.battery_health,
+    RoutineType.cpu_cache,
+    RoutineType.dns_resolution,
+    RoutineType.memory,
+    RoutineType.power_button,
+    RoutineType.sensitive_sensor,
+  ];
+
+  get error() { return this._error; }
+  get visibleRoutines() {
+    let routines;
+    if (this._availableRoutines) {
```

Comment:
I put the <code>routines = this._visibleRoutines.filter...</code> part of code in the <code>else</code> block since there is a circumstance that the <code>getAvailableRoutines()</code> fails and the value of <code>this.availableRoutines()</code> is <code>undefined</code>. And in that case, an error will occur since it cannot read the property <code>includes</code> of undefined.